ADApp/ADSrc/NDArray.h

```diff
@@ -124,6 +126,26 @@ class epicsShareClass NDArray {
     NDAttributeList *pAttributeList;  /**< Linked list of attributes */
 };
 
+// This class defines the object that is contained in the std::multilist for sorting NDArrays in the freeList_.
+// It defines the < operator to use the NDArray::dataSize field as the sort key
+
+// We would like to hide this class definition in NDArrayPool.cpp and just forward reference it here.
+// That works on Visual Studio, and on gcc if instantiating plugins as heap variables with "new", but fails on gcc
+// if instantiating plugins as automatic variables.
+//class sortedListElement;
+
+class freeListElement {
```

Comment:
Could we have a private default constructor to ensure we aren't accidentally creating invalid objects when using them with std::multiset (or any other container)?

Reply:
I have made the freeListElement constructor private, and made this a friend of the NDArrayPool class.

Reply:
We actually meant the "default constructor" which is a constructor that takes no arguments and the compiler will implicitly define it if we do not do it. This implicitly defined constructor will not initialise the variables properly and the object will be invalid. If we declare it private then we are prevented from accidentally create objects like that.

https://en.cppreference.com/w/cpp/language/default_constructor

Reply:
OK, I understand.  I have made the non-default constructor public and the default constructor private.